### PR TITLE
X.pdf(x) for invalid x in DieDistribution raises ValueError

### LINKED ISSUE #10075: X.pdf(x) for Symbol x returns 0
```
>>> x = Symbol('x')
>>> from sympy.stats.frv_types import DieDistribution
>>> X = DieDistribution(6)
>>> X.pdf(x)
0
```
Either this should raise a `ValueError` or unevaluated. Returning unevaluated seems more favourable.

### PR DESCRIPTION
**Pull request by gxyd**
fixes #10075

### CODE REVIEW

**Comment by gxyd** [comment]
I was hoping that something like
```
>>> x = Symbol('x')
>>> X = DieDistribution(6)
>>> X.pdf(x)
PDF(X, x)    # some PDF be returned though i guess no such PDF  class or function is there
```
I expected something similar to `Contains`
```
>>> A.contains(x)
Contains(A, x)
```

**Comment by jksuom** [comment]
I think it could be possible to return a sensible ``X.pdf(x)`` even for a symbol ``x``. For an ordinary die (with 6 faces) it would be ``AddWithLimits(KroneckerDelta(x, i)/6, (i, 1, 6))`` (though that looks a bit clumsy).

**Comment by gxyd** [comment]
It looks good to me. I did not know about the `KroneckerDelta` function till now though.
> AddWithLimits(KroneckerDelta(x, i)/6, (i, 1, 6)) (though that looks a bit clumsy).

I think i am ok, with that.

**Comment by jksuom** [comment]
I now think that we should have ``Sum`` instead of ``AddWithLimits``. I had forgotten its existence. It seems that ``X.pdf(x).subs(x, 1).doit()`` would compute correctly with ``Sum`` since it has its own version of ``doit``. There could be some tests for such substitutions. And maybe for some illegal substitution as well.

**Comment by gxyd** [comment]
I myself, even before seeing with `AddWithLimits` checked with `Sum.doit()` . But i could not though make that out how to do that. But Ok, let me try again. I may now be able to come to solution.

**Comment by gxyd** [comment]
I have made the changes. Perhaps previously i was also using `.doit()` itself in code. Now using ony `Sum` works fine.

**Comment by jksuom** [comment]
This looks good to me. The separation of arguments that do not belong to the probability space (i.e. integers between 1 and ``sides``) into two classes: numbers returning 0, and non-numbers returning an error, is somewhat arbitrary, but I cannot think of a better way now.

**Comment by jksuom** [comment]
I'll merge in a few hours if no-one objects.
